syntaxes/notes.tmLanguage.json

```diff
@@ -29,6 +29,18 @@
           "name": "support.function.text.markdown.notes.tag.title"
         }
       }
+    },
+    {
+      "match": "(?<= |,|^)(\\@)([\\p{L}\\d\\-_]+)",
```

Comment:
Looking at this again -- I think this would match, for example, a string like `@jordan23` -- am I correct in reading your description of the bibtex notation syntax that the string will always end in a 4 digit year? 

> (eg `@author_title_year`)

If so I am thinking we should update this regex (and the others) to have a `_\\d{4}` at the end of the word?

Reply:
Hm this could be a problem, because the citekey can have an arbitrary format (eg `@author_title` or even just `@author`) and different people use different styles (I just happen to prefer `@author_title_year`). This means the citekey is generally ambiguous between pandoc-style citation and github/twitter handles. Even the [tests](https://github.com/kortina/vscode-markdown-notes/pull/124/files#diff-2c44fae7f6c4a0d394637ec0f4f2b4b89dd3b0a7f76ee6789bec2b938a1478a2R216) I added assume it can be of `@string` format. I see two ways to resolve the worry you raised:

1. We limit citations to the stricter form, where they need to appear inside square brackets only, eg `[@reference]` instead of just `@reference`. This is the most commonly used style in pandoc as shown [here](https://pandoc.org/MANUAL.html#extension-citations). Pandoc does however support citations outside of brackets as well (and so does Markdown Preview Extended) so we would be restricting that functionality.
2. Accept that `@reference` will be ambiguous between citations and handles with the following consequences:
    - syntax highlighting will always work even for github/twitter handles (maybe thats actually a good thing?)
    - if no global bibtex file is set, no parsing, autocompletion or go-to definition will work on handles
    - if global bibtex file is set and one markdown file contains both handles and citations, autocomplete will be suggesting references instead of usernames (but one can just ignore those suggestions) and go-to definition will only work on citations, whereas it will show the standard "No definition found" on handles

What do you think about these two options?

Reply:
I like something along the lines of your (2).

I don't think we can conditionally do syntax highlighting, since this is governed by `syntaxes/notes.tmLanguage.json` not runtime code, but perhaps we could just document how the syntax highlighting works and have an FAQ like, "How can I disable syntax highlighting for `@words` / BibTex citations?"

https://github.com/kortina/dotfiles/blob/76780a8732d4af955a361b80022295e6650b85c3/vscode/settings.json#L347

Parts (b) and (c) of your (2) seem good. Can you remind me -- is that how the current implementation works?

If so, I think we can just merge this in and then async you could send over an FAQ commit either as part of this PR ( [here](https://github.com/kortina/vscode-markdown-notes/blob/master/TROUBLESHOOTING-FAQ.md) ) or as a comment here if this merges in before writing it up and then I can add it to the FAQ.

Reply:
Yes 2b and 2c are already working this way at the moment and I've now added a section into FAQ about overriding the highlight. Is [this](https://github.com/kortina/vscode-markdown-notes/pull/124/files#diff-9c44577fe2682f5ef992dff912c37c17513ab7b898a0b0b7ce5e035cc9ec1e5eR50) what you had in mind?